shared-actions/secrets-scan/action.yml

```diff
@@ -0,0 +1,27 @@
+name: Secrets Scan
+description: 'Scan for secrets using Gitleaks'
+inputs:
+  docker-username:
+    description: 'Docker Username'
+    required: true
+  docker-password:
+    description: 'Docker Password'
+    required: true
+  docker-registry:
+    description: 'Docker Registry'
+    required: true
+  gh-token:
+    description: 'Github Token'
+runs:
+  using: "composite"
+  steps:
+    - run: docker login -u=${{ inputs.docker-username }} -p=${{ inputs.docker-password }} ${{ inputs.docker-registry}}
+      shell: bash
+
+    - run: ${{ github.action_path }}/secrets-scan.sh
```

Comment:
I'm interested to know why would we need this action path?

Reply:
This points to the folder where the code for the action is, so if we ever move the action elsewhere (e.g. its own separate repo) it would be pointing at path for the script. We saw it in the example: https://docs.github.com/en/actions/creating-actions/creating-a-composite-run-steps-action#creating-an-action-metadata-file

Reply:
will this work outside this repo for other repos without provinding that environment variable?

Reply:
This one is not an env variable or an input, but a variable from Github itself, so it should work from any repo.

Reply:
wouldn't this called from another repo, provide another path?

Reply:
From my understanding of the docs, it's always pointing to the path of the action in the original repo where the action lives, so it's useful to point at other files that are part of the action (like support scripts like in this case). It's only available for the composite step action type that we are using here https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context

Reply:
Nice! 
In any case we'd need to try this afterwards!